### ISSUE TITLE
Update for new repository format

### ISSUE THREAD

**Issue by SammysHP** [SammysHP/anduril-buildenv-docker#1]
Anduril was moved into a new repository, using a different code structure and new build scripts.

https://github.com/ToyKeeper/anduril

**Comment by EpochalEngineer** [comment]
In case you happen to still be working on this, getting some linker errors. Did need to run `./make dfp` as described in the new anduril repo and error message to get to this point:

```
bash-5.1# ./make sofirn-sp10-pro
bin/version-string.sh: line 34: git: command not found
bin/version-string.sh: line 50: git: command not found
===== anduril 0. : sofirn-sp10-pro =====
/usr/lib/gcc/avr/10.3.0/../../../../avr/bin/ld: address 0x80382a of ui/anduril/anduril.elf section `.data' is not within region `data'
/usr/lib/gcc/avr/10.3.0/../../../../avr/bin/ld: address 0x8038ba of ui/anduril/anduril.elf section `.bss' is not within region `data'
/usr/lib/gcc/avr/10.3.0/../../../../avr/bin/ld: address 0x80382a of ui/anduril/anduril.elf section `.data' is not within region `data'
/usr/lib/gcc/avr/10.3.0/../../../../avr/bin/ld: address 0x8038ba of ui/anduril/anduril.elf section `.bss' is not within region `data'
collect2: error: ld returned 1 exit status
ERROR: build failed
```

Building outside of docker with no changes is successful

**Comment by SammysHP** [comment]
It worked perfectly fine with the `new-build-system` branch, except that git is missing and that dd targets don't work without the DFP.

```
===== anduril 0. : sofirn-sp10-pro =====
  Device: attiny1616
  Program:    9628 bytes (58.8% Full)
  Data:        186 bytes (9.1% Full)
  # 509c1fc89e595a7377678605956e40c6
  > hex/anduril.sofirn-sp10-pro.hex
…
===== anduril 0. : thefreeman-avr32dd20-devkit =====
avr-cpp: fatal error: cannot read spec file 'device-specs/specs-avr32dd20': No such file or directory
compilation terminated.
ERROR: build failed
…
===== 71 builds succeeded, 1 failed =====
FAIL: thefreeman-avr32dd20-devkit
```

For now I had to create `arch/dfp/{attiny,avrdd}` directories (can be empty) because the make script checks for their existence. But my Docker image is designed to work without the "proprietary" DFP. It's possible that the old version of the compiler doesn't work well together with the official DFP.

In short:

- Create `arch/dfp/{attiny,avrdd}`
- Run `docker run --rm -v /path/to/source/repo:/src anduril-builder sofirn sp10-pro`

**Comment by SammysHP** [comment]
Update: Added git to the image.

Also confirmed that the DFP isn't compatible with avr-gcc 10.3.0 that was chosen for "best" compatibility with Anduril and avr-libc (creating the smallest hex files).

See ToyKeeper/anduril#32 for a method to disable the DFP check so that it can be built without the DFP.

**Comment by EpochalEngineer** [comment]
Sorry for not being clear. I was using that branch as the commit was referenced here. So when I encountered the missing DFP file messages I ran the script to grab the proprietary DFP to get you those error messages.

So essentially you did what I attempted to describe and verified that gcc incompatibility. Thanks for the context on why that version was shown and how to build without DFP
